cores/esp8266/core_esp8266_noniso.cpp

```diff
@@ -40,75 +41,9 @@ char* ultoa(unsigned long value, char* result, int base) {
 }
 
 char * dtostrf(double number, signed char width, unsigned char prec, char *s) {
-    bool negative = false;
-
-    if (isnan(number)) {
-        strcpy(s, "nan");
-        return s;
-    }
-    if (isinf(number)) {
-        strcpy(s, "inf");
-        return s;
-    }
-
-    char* out = s;
-
-    int fillme = width; // how many cells to fill for the integer part
-    if (prec > 0) {
-        fillme -= (prec+1);
-    }
-
-    // Handle negative numbers
-    if (number < 0.0) {
-        negative = true;
-        fillme--;
-        number = -number;
-    }
-
-    // Round correctly so that print(1.999, 2) prints as "2.00"
-    // I optimized out most of the divisions
-    double rounding = 2.0;
-    for (uint8_t i = 0; i < prec; ++i)
-        rounding *= 10.0;
-    rounding = 1.0 / rounding;
-
-    number += rounding;
-
-    // Figure out how big our number really is
-    double tenpow = 1.0;
-    int digitcount = 1;
-    while (number >= 10.0 * tenpow) {
-        tenpow *= 10.0;
-        digitcount++;
-    }
-
-    number /= tenpow;
-    fillme -= digitcount;
-
-    // Pad unused cells with spaces
-    while (fillme-- > 0) {
-        *out++ = ' ';
-    }
-
-    // Handle negative sign
-    if (negative) *out++ = '-';
-
-    // Print the digits, and if necessary, the decimal point
-    digitcount += prec;
-    int8_t digit = 0;
-    while (digitcount-- > 0) {
-        digit = (int8_t)number;
-        if (digit > 9) digit = 9; // insurance
-        *out++ = (char)('0' | digit);
-        if ((digitcount == prec) && (prec > 0)) {
-            *out++ = '.';
-        }
-        number -= digit;
-        number *= 10.0;
-    }
-
-    // make sure the string is terminated
-    *out = 0;
+    char fmt[32];
+    sprintf(fmt, "%%%d.%df", width, prec);
```

Comment:
Might need to be labelled as breaking? See
https://github.com/arendst/Tasmota/blob/1598e6227a31be5f27081c5a660d22d21613c0ca/platformio.ini#L83
https://github.com/arendst/Tasmota/blob/development/pio/strip-floats.py
https://github.com/xoseperez/espurna/blob/dev/code/scripts/espurna_utils/float_support.py

It removes `-u _printf_float` from linker flags, so this might come as a surprise b/c code uses dtostrf routinely :)

Reply:
Ah, crap.  Thank you for reminding me about that!  I had a niggle about it, but it's been maybe 2 years since the whole -u print_float stuff.

I'll open a new issue.  The solution is not clear to me now.